renderlib/AxisHelperTool.cpp

```diff
@@ -0,0 +1,128 @@
+#include "AxisHelperTool.h"
+
+struct ManipColors
+{
+  static constexpr glm::vec3 xAxis = { 1.0f, 0.0f, 0.0f };
+  static constexpr glm::vec3 yAxis = { 0.0f, 1.0f, 0.0f };
+  static constexpr glm::vec3 zAxis = { 0.0f, 0.0f, 1.0f };
+  static constexpr glm::vec3 bright = { 1.0f, 1.0f, 1.0f };
+};
+
+void
+AxisHelperTool::action(SceneView& scene, Gesture& gesture)
+{
+}
+
+void
+AxisHelperTool::draw(SceneView& scene, Gesture& gesture)
+{
+  if (!scene.scene) {
+    return;
+  }
+  if (!scene.scene->m_showAxisHelper) {
+    return;
+  }
+
+  const glm::vec2 resolution = scene.viewport.region.size();
+  // Move tool is only active if something is selected
+  if (!scene.anythingActive()) {
+    return;
+  }
+
+  LinearSpace3f camFrame = scene.camera.getFrame();
+
+  AffineSpace3f target;
+  // assume camera target is center of camera view!!
+  target.p = scene.camera.m_Target;
+
+  glm::vec3 viewDir = (scene.camera.m_From - target.p);
+
+  AffineSpace3f axis;
+  axis.p = target.p;
+
+  // translate this in screen space toward the lower left corner.
+  // note that this is still drawn in perspective and will have fov distortion.
+  // Ideally we want to give this its own viewport as if it were centered.
+  // (Or translate the center of projection!)
+  static constexpr float XPOS_NDC = -0.45f;
+  static constexpr float YPOS_NDC = -0.45f;
+
+  // Draw the manipulator to be at some constant size on screen
+  float scaleFactor = 0.333; // make it smaller
+  float scale = scaleFactor;
```

Comment:
why do we set `scale` to `scaleFactor` here?

Reply:
This is just to initialize it to something.  The if/else below is going to reset it but I wanted to give it some valid init value here.